diff --git a/ext/openexr/Makefile.am b/ext/openexr/Makefile.am
index 1a95fee6b..6a21d8522 100644
--- a/ext/openexr/Makefile.am
+++ b/ext/openexr/Makefile.am
@@ -6,7 +6,11 @@ libgstopenexr_la_CFLAGS = \
 	$(OPENEXR_CFLAGS)
 libgstopenexr_la_CXXFLAGS = \
 	$(GST_PLUGINS_BASE_CFLAGS) $(GST_BASE_CFLAGS) $(GST_CXXFLAGS) \
-	$(OPENEXR_CFLAGS) -std=c++98
+	$(OPENEXR_CFLAGS)
+#libgstopenexr_la_CXXFLAGS = \
+#	$(GST_PLUGINS_BASE_CFLAGS) $(GST_BASE_CFLAGS) $(GST_CXXFLAGS) \
+#	$(OPENEXR_CFLAGS) -std=c++98
+
 libgstopenexr_la_LIBADD = \
 	$(GST_PLUGINS_BASE_LIBS) -lgstvideo-$(GST_API_VERSION) \
 	$(GST_BASE_LIBS) $(GST_LIBS) $(OPENEXR_LIBS)
