diff --git a/gst/vaapi/gstvaapivideomemory.c b/gst/vaapi/gstvaapivideomemory.c
index 16bf143..6f6a45a 100644
--- a/gst/vaapi/gstvaapivideomemory.c
+++ b/gst/vaapi/gstvaapivideomemory.c
@@ -1070,11 +1070,22 @@ gst_vaapi_dmabuf_memory_new (GstAllocator * base_allocator,
    * GST_FD_MEMORY_FLAG_NONE. So when being freed it calls close on the fd
    * because GST_FD_MEMORY_FLAG_DONT_CLOSE is not set. */
   dmabuf_fd = gst_vaapi_buffer_proxy_get_handle (dmabuf_proxy);
-  if (dmabuf_fd < 0 || (dmabuf_fd = dup (dmabuf_fd)) < 0)
+  // kl: original 202002202139
+  // if (dmabuf_fd < 0 || (dmabuf_fd = dup (dmabuf_fd)) < 0)
+ //   goto error_create_dmabuf_handle;
+
+  //mem = gst_dmabuf_allocator_alloc (base_allocator, dmabuf_fd,
+  //    gst_vaapi_buffer_proxy_get_size (dmabuf_proxy));
+
+  if (dmabuf_fd < 0)
     goto error_create_dmabuf_handle;
 
-  mem = gst_dmabuf_allocator_alloc (base_allocator, dmabuf_fd,
-      gst_vaapi_buffer_proxy_get_size (dmabuf_proxy));
+     mem = gst_fd_allocator_alloc (base_allocator, dmabuf_fd,
+     gst_vaapi_buffer_proxy_get_size (dmabuf_proxy), GST_FD_MEMORY_FLAG_DONT_CLOSE);
+ //kl: end 202002202139
+
+//  mem = gst_dmabuf_allocator_alloc (base_allocator, dmabuf_fd,
+//      gst_vaapi_buffer_proxy_get_size (dmabuf_proxy));
   if (!mem)
     goto error_create_dmabuf_memory;
 
